Refactor packet encoding to use pmmp/ext-encoding

flovrian opened the pull request:
After reading through the changes you requested on the other pull request, I decided to propose these. I hope they're to your liking!
I used a SignedLong here: `VarInt::writeSignedLong($this->injection, -1);` — to my understanding, a negative value can’t be unsigned. If this was intentional (e.g., required by the Bedrock protocol), feel free to request changes.

flovrian committed:
Update InjectingData.php

Refactor packet encoding to use pmmp/ext-encoding

flovrian committed:
Update UpdateSubChunkBlocksInjector.php

Refactor packet encoding to use pmmp/ext-encoding

flovrian committed:
Update UpdateSubChunkBlocksInjector.php

Refractor curly brackets

platz1de commented on src/platz1de/EasyEdit/world/blockupdate/InjectingData.php:
```diff
@@ -2,42 +2,45 @@
+		$serializer = new ByteBufferWriter();
+        VarInt::writeSignedInt($serializer, $this->position->getX());
+        VarInt::writeUnsignedInt($serializer, Binary::unsignInt($this->position->getY()));
+        VarInt::writeSignedInt($serializer, $this->position->getZ());
```
This can use `pocketmine\network\mcpe\protocol\serializer\CommonTypes::putBlockPosition`

platz1de reviewed (commented):
The actor value is actually supposed to be unsigned (-1 would be encoded as 0xffffffff), but I couldn't figure out where I got that -1 from.
Pocketmine [defaults to 0 when no actors are involved](https://github.com/pmmp/BedrockProtocol/blob/master/src/types/UpdateSubChunkBlocksPacketEntry.php#L34), so it's probably best to use 0 instead of -1 here

flovrian committed:
Update InjectingData.php

Use putBlockPosition() instead of manually encoding block positions and change value used when no actors are involved.

flovrian reviewed (commented):
Should be fine now, no?

platz1de committed:
Formatting

Signed-off-by: platz1de <51201131+platz1de@users.noreply.github.com>

platz1de commented:
Thank you!
